### ISSUE TITLE
Cursor unit test: check that cursor reached the end

### ISSUE THREAD

**Issue by matt-uib** [opensheetmusicdisplay/opensheetmusicdisplay#864]
For a given demo musicxml file:
Do osmd.cursor.next() in a loop 
until (osmd.cursor.Iterator.EndReached === true)
to see that there are no exceptions occurring on the way through the piece. 

Do this test for all our demo pieces.

**Comment by matt-uib** [comment]
@fredmcgroarty would be great if you can try this one!

**Comment by sschmidTU** [comment]
Probably include a fail condition like only calling cursor.next() 10 000 times (should still be fast enough), so that the tests don't go into an infinite loop.
(if 10  000 spaces isn't enough for one of our samples, we'll find that out very quickly)

**Comment by sschmidTU** [comment]
I just checked that the cursor works without calling osmd.render().
Because calling render on all our samples would take way too long for unit tests i guess, and it's not so easy to do in unit tests anyways, because you would need to fake the browser elements, like in `generateImages_browserless.js`.

The moral is: you don't need to call osmd.render() in these tests ;)

**Comment by sschmidTU** [comment]
But osmd.load() is of course necessary for each sample.

**Comment by YolandaFeng1995** [comment]
how can the cursor move back when it encounters repeat\coda etc.?

**Comment by loganknecht** [comment]
I too would like to hear suggestions on how to repeat sections with the cursor.

**Comment by sschmidTU** [comment]
It should soon be in the OSMD main repo, as it comes with the audio player, which is in early access right now (Github Sponsors).

**Comment by loganknecht** [comment]
@sschmidTU Total side question, but I've been a huge fan of this project for a long time, but I don't have the means to sponsor at the rates you all have. Is there any mechanism for donating $5 a month to this?

**Comment by sschmidTU** [comment]
@loganknecht that can be done via Paypal, see link at the bottom of our main page:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay

We would really appreciate the support! :)

**Comment by YolandaFeng1995** [comment]
> @loganknecht that can be done via Paypal, see link at the bottom of our main page:
> https://github.com/opensheetmusicdisplay/opensheetmusicdisplay
> 
> We would really appreciate the support! :)
Any other ways to be a sponsor? something wrong with the current means... oops

**Comment by sschmidTU** [comment]
@YolandaFeng1995 what do you mean? Both Github Sponsors and Paypal seem to work, to my knowledge. These are the two ways (and obviously we recommend Github Sponsors because you get rewards for it)

**Comment by YolandaFeng1995** [comment]
> @YolandaFeng1995 what do you mean? Both Github Sponsors and Paypal seem to work, to my knowledge. These are the two ways (and obviously we recommend Github Sponsors because you get rewards for it)

emm... 1.no Paypal account. 2.my card was told invalid (in use in fact) when I tried.

**Comment by YolandaFeng1995** [comment]
can the cursor do the repetition iterator if I modify some parameters of  MusicPartManagerIterator?  or any other suggestions?

**Comment by sschmidTU** [comment]
@YolandaFeng1995 no, it's not that easy. You'll have to wait until the repeats are merged into OSMD (hopefully soon).
The audio player has it built in, so if you sponsor OSMD, you can see it in the early access audio player build.

**Comment by YolandaFeng1995** [comment]
Hi～ how can i get access to the audio player?

**Comment by sschmidTU** [comment]
@YolandaFeng1995 We're currently preparing the audio player and other sponsor builds, it should be available very soon!

**Comment by sschmidTU** [comment]
@YolandaFeng1995 I invited you to the audio player repository a long time ago. Did you not see the invitation?
